utils/resolve.go

```diff
@@ -0,0 +1,56 @@
+package utils
+
+import (
+	"bufio"
+	"os"
+	"regexp"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// ExtractDNSServerFromResolvConf takes a list of filenames
+// the files are supposed to be formated in the resolv.conf format.
+// it will extract ipv4 DNS server addresses and return these
+func ExtractDNSServerFromResolvConf(filenames []string) ([]string, error) {
```

Comment:
I would suggest the func to deal with an io.ReadCloser (file) instead of a list of file paths.
That way we can write a unit test easily for it and then in the caller we will range over the provided paths and catenate the extracted results.

Reply:
Went down the road of io.fs which allows for a MapFS to be handed to the tests.